Update test brief (#1)
dave-perry/tech-test-ios: README.md

Please complete the task detailed in the brief below. 

> We appreciate that your free time is important, and we expect this to take no longer than an evening (2-3 hours) to complete. If you feel your code is unfinished please leave some notes in your README.md explaining what you would do next, given more time. Please take note of the [How We Score](#How-We-Score) section below to ensure that you have demonstrated everything that we are looking for.

Reviewer: > we expect this to take no longer than an evening (2-3 hours) to complete

This might imply that we expect you to _fully_ complete the task, which isn't really what we mean, I don't think... How about something like `we want you to work on this for no longer than an evening (2-3 hours)`?

Author: With the updated brief & marking criteria, as well as the coming project template I would expect that to be the case.

Reviewer: Ah yeah makes sense I guess, hopefully we can get some confidence from internal testing that that's not unrealistic

Author: Yeah - hoping the internal testing will validate this claim!

We are looking purely at your technical skill in this exercise. Subsequent interview stages will cover the other aspects that we believe are also important.

# How We Score

When reviewing your project, we'll score it on:

> - How much of the task has been completed and how accurately it meets the brief (a following stage in the interview process requires a certain level of functionality)

Reviewer: ```suggestion
- How much of the task has been completed and how accurately it meets the brief. We will continue this project at a later stage of the interview process, so the functionality we've asked for needs to be present and work as intended
```
Suggested rewording to hopefully make it even clearer what this means

- Approach and design patterns used to enable testing. We're looking for code to be written in a way to enable testability and a strong demonstration of how you would test at least one on your created classes/structs in full - the less trivial the better

> - Any bugs that are discovered

Reviewer: ```suggestion
- Any bugs we find in your code
```
Trivial but just reworded as the passive voice could potentially be ambiguous/misleading

> - Approach and design patterns used to enable testing. We're looking for code to be written in a way to enable testability and a strong demonstration of how you would test at least one on your created classes/structs in full - the less trivial the better

Reviewer 2: ```suggestion
- Approach and design patterns used to enable testing. We're looking for code to be written in a way to enable testability and a strong demonstration of how you would test at least one of your created classes/structs in full - the less trivial the better
```

- SwiftUI usage
- How clean, modular and extensible the code is
- The applied architecture and how the code is structured
- How you handle errors. The customer should be made aware of any critical issues
- Use of modern technologies such as Combine or async/await (where appropriate)
- Any information provided in your README
- Source control management - please commit regularly and include the .git folder in your submission

We will not consider:

> - How well the UI matches the wireframe below. It is shown for illustration only

Reviewer: "We won't consider whether your UI matches the wireframe you're building against" feels inaccurate to me. We will consider it to the point that you've created a scrolling list of cards (which is what the wireframe shows), _but not beyond that_... I think this bit is really hard to word without just straight-out saying "Don't focus on the styling of the UI".

I wonder if we should also update the wireframe to not have a masonry-style layout - nobody ever does it anyway, it's not really practical in the time given the other things we want them to focus on, so maybe it's weird to have it there? In fact, maybe we could provide an actual (very simple) design, rather than a wireframe, if it's confusing to clarify what we're after? After all the candidate will rarely, if ever, be asked to build against wireframes in the role...

Author: Should we just remove the wireframe image? Replace it with some words that clarify the expectation is a scrolling list/grid of card images?

Reviewer: True, that might get across "we really don't care if this is pretty" even more so than the wireframe does. But I think a simple design might work better, since we can't really be any clearer than that about expectations, plus it matches how we usually work in reality (at least ideally!) We could A/B test with Kostia and Artem 😂  Would be good to see what others think about this.

Author: Happy to swap this out 👍

> - Extra scope or functionality not requested for the task

Reviewer: 👏
